### PR TITLE
Weighted moving average

### PR DESCRIPTION
**Pull request by SiggiSmara**
Preemptively creating the pull request for the weighted moving average. More explanation in #53

### CODE REVIEW

File: R/smoothingFilters-functions.R [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+##
+## returns:
+##  double
+##
+.movingWeightedAverage<-function(y, halfWindowSize=2L) {
+  windowSize <- 2L * halfWindowSize + 1L
+  myweigths<-1/2**abs(-halfWindowSize:halfWindowSize)
+  myweigths<-myweigths/sum(myweigths)
+  .filter(y, hws=halfWindowSize,
+          coef=matrix(myweigths, nrow=windowSize, ncol=windowSize, byrow = TRUE))
```
Comment (sgibb) [inline R/smoothingFilters-functions.R]: Please use `byrow=TRUE` instead of `byrow = TRUE`.

**Comment by SiggiSmara** [comment]
No problem with the coding style. I shall try to be more careful in the future. I think your suggestion of incorporating the weights into the MovingAverage is a very good one. This also lessens the changes to be added to packages that rely on this function. I've made the changes, also to the unit tests and man pages.

File: man/smoothIntensity-methods.Rd [hunk 2]
```diff
@@ -26,7 +27,12 @@ This method smoothes the intensity values of a
     selected \code{method}.}
   \item{\dots}{arguments to be passed to \code{method}. \code{SavitzkyGolay} has
     an additional \code{polynomialOrder} argument (default: \code{3}) to control
-    the order of the filter. Unused for \code{MovingAverage}}
+    the order of the filter. Unused for \code{MovingAverage}.
```
Comment (sgibb) [inline man/smoothIntensity-methods.Rd]: `\dots` isn't "unused" in MovingAverage anymore. This sentence could be removed.

File: R/smoothingFilters-functions.R [hunk 3]
```diff
@@ -1,21 +1,26 @@
...
   .stopIfNotIsValidHalfWindowSize(halfWindowSize, n=length(y))
   windowSize <- 2L * halfWindowSize + 1L
+  if (weighted) {
+    weights <- 1 / 2^abs(-halfWindowSize:halfWindowSize)
+  } else {
+    weights <- rep.int(1L, windowSize)
+  }
   .filter(y, hws=halfWindowSize,
-          coef=matrix(1L / windowSize, nrow=windowSize, ncol=windowSize))
+          coef=matrix(weights/sum(weights), nrow=windowSize, ncol=windowSize, byrow=TRUE))
```
Comment (sgibb) [inline R/smoothingFilters-functions.R]: Could you please add spaces around `/` (I know I haven't done it myself in my last comment, sry).

File: tests/testthat/test_smoothingFilters-functions.R [hunk 4]
```diff
@@ -3,11 +3,15 @@ context("smoothingFilters")
 test_that(".movingAverage", {
   values <- c(rep(3, 3), 4:7, rep(8, 3))
   expect_equal(MALDIquant:::.movingAverage(1:10, halfWindowSize=2), values)
+  expect_equal(MALDIquant:::.movingAverage(1:10, halfWindowSize=2, 
+    weighted=TRUE), values)
 
   values <- c(rep(4, 4), 5:6, rep(7, 4))
   expect_equal(MALDIquant:::.movingAverage(1:10, 3), values)
+  expect_equal(MALDIquant:::.movingAverage(1:10, 3, weighted=TRUE), values)
```
Comment (sgibb) [inline tests/testthat/test_smoothingFilters-functions.R]: Could you please add an unit test where the default and the weighted version differ?

File: man/smoothIntensity-methods.Rd (hunk 2 above)
Comment (SiggiSmara) [reply man/smoothIntensity-methods.Rd]: Sorry.. missed that one. Done.

File: R/smoothingFilters-functions.R (hunk 3 above)
Comment (SiggiSmara) [reply R/smoothingFilters-functions.R]: No problem. Done.

File: tests/testthat/test_smoothingFilters-functions.R (hunk 4 above)
Comment (SiggiSmara) [reply tests/testthat/test_smoothingFilters-functions.R]: Good idea. Done.

**Comment by sgibb** [comment]
Thanks a lot!

**Pull request merged**